surveys/promise-rejections/survey.md

```diff
@@ -0,0 +1,235 @@
+# Node.js Promise reject use case survey
+
+Today, Node.js handles unhandled rejections by emitting a deprecation warning to stderr. The warning shows the stack where the rejection happened, and states that in future Node.js versions unhandled rejections will result in Node.js exiting with non-zero status code. We intend to remove the deprecation warning, replacing it with a stable behavior which might be different from the one described on the deprecation warning. We're running this survey to better understand how Node.js users are using Promises and how they are dealing with unhandled rejections today, so we can make an informed decision on how to move forward.
+
+## What is a Promise rejection?
+
+A Promise rejection indicates that something went wrong while executing a Promise or an async function. Rejections can occur in several situations: throwing inside an async function or a `Promise` executor/then/catch/finally callback, when calling the `reject` callback of an `executor`, or when calling `Promise.reject`.
+
+```js
+Promise.reject(new Error())  // This will result in a rejection
+
+new Promise((fulfill, reject) => {
+  reject(new Error());  // This will result in a rejection
+});
+
+new Promise(() => {
+  throw new Error();  // This will result in a rejection
+})
+
+Promise.resolve().then(() => {
+  throw new Error();  // This will result in a rejection
+})
+
+Promise.reject().then(() = {}, () => {
+  throw new Error();  // This will result in a rejection
+}
+
+Promise.reject().catch(() => {
+  throw new Error();  // This will result in a rejection
+})
+
+Promise.resolve().finally(() => {
+  throw new Error();  // This will result in a rejection
+});
+
+async function () {
+  throw new Error();  // This will result in a rejection
+}
+
+async function () {
+  if (a === undefined)
+    a();  // This will result in a rejection
+}
+```
+
+Adding the `async` keyword to a function will turn any exceptions thrown inside that function (or any throw propagated from other functions called within it) into a rejection. The same happens when refactoring callback based code that throws into async functions / Promises. Below is an example of a callback based code refactored to Promises where exceptions become rejections:
```

Comment:
This paragraph says that the example is below, but the example appears to be missing…?

Reply:
I think something went wrong during a rebase, the example was indeed supposed to be below, but I can't find the example anymore :facepalm:

I'll rewrite the example.